### PR TITLE
fix: Default length of mock string to 1 while HCL parsing

### PR DESCRIPTION
**Pull request by fatihtokus**

### CODE REVIEW

**Comment by aliscott** [comment]
Thanks @fatihtokus 🙏. Let us know when this is ready for review.

**Comment by aliscott** [comment]
Thanks @fatihtokus 🙏. Is this ready for review? It looks like it might need rebased against master.

**Comment by fatihtokus** [comment]
@aliscott, ready for review

**Comment by hugorut** [comment]
hey @fatihtokus, thanks for this PR. Before I look at it, could you remove all the comment formatting from the changes, this is just noise, and it helps to isolate on the actual changes you've made.

**Comment by fatihtokus** [comment]
@hugorut , reverted unnecessary changes, but the tests are failing with: Error initializing Terraform working directory: exec: "terraform": executable file not found in %PATH%

File: internal/hcl/funcs/collection.go [hunk 1]
```diff
@@ -637,6 +638,10 @@ var MapFunc = function.New(&function.Spec{
 // Length returns the number of elements in the given collection or number of
 // Unicode characters in the given string.
 func Length(collection cty.Value) (cty.Value, error) {
+	if cty.String.Equals(collection.Type()) && collection.IsKnown() && !collection.IsMarked() && strings.HasSuffix(collection.AsString(), "-mock") {
+		return cty.NumberIntVal(1), nil
+	}
```
Comment (hugorut) [inline internal/hcl/funcs/collection.go:643]:
```suggestion
    var s string
    _ := gocty.FromCtyValue(collection, &s)
	if strings.HasSuffix(s, "-mock") {
		return cty.NumberIntVal(1), nil
	}
```

**Review by hugorut — COMMENTED** [review]
Thanks @fatihtokus, this looks good and returning length 1 is a sensible default for mock string values.

However, it's not really a fix for https://github.com/infracost/infracost/issues/1856, as this doesn't handle the underlying problem of trying to infer the collection types from `data` or unresolvable values. This requires more work so that mock values work with Terraform functions that require collection types. That being said, the Infracost team will handle this work internally, as it's quite a big change to how we do mock types.

In order to get this accepted, can we add an HCL test under `internal/hcl/parser_test.go` to ensure this functionality is covered by the parser?

Thanks for your contribution

**Comment by hugorut** [comment]
Hey @fatihtokus, any updates on this PR? We've got a release going out soon so it would be great to try and get it into that. Thanks

**Comment by fatihtokus** [comment]
> Hey @fatihtokus, any updates on this PR? We've got a release going out soon so it would be great to try and get it into that. Thanks

Hi @hugorut, I was busy last week. Hopefully this week will have a looks

**Pull request closed without merging**